build.gradle

```diff
@@ -622,3 +625,19 @@ allprojects {
     }
   }
 }
+
+allprojects {
+  task checkPart1
+  task checkPart2 
+  tasks.matching {it.name == "check"}.all { task ->
```

Comment:
nit: I would not name the local "task" since that masks the keyword of the same name